test/services/NotProcessedFilesServiceSpec.scala

```diff
@@ -0,0 +1,230 @@
+      when(mockFileNotificationRepository.getFilesReceivedBySDES()).thenReturn(Future.successful(Seq.empty))
+      val result: Either[ScheduleStatus.JobFailed, String] = await(service.invoke)
+      result.isRight shouldBe true
+      result.getOrElse("fail") shouldBe "Processed all notifications"
+    }
+
+    "process the notifications and return Right if they all succeed - only process if nextAttemptAt is < now" in new Setup {
+      when(mockFileNotificationRepository.getPendingNotifications()).thenReturn(Future.successful(pendingNotifications))
+      when(mockFileNotificationRepository.getFilesReceivedBySDES()).thenReturn(Future.successful(pendingNotifications))
+      when(mockFileNotificationRepository.updateFileNotification(Matchers.any(), Matchers.any())).thenReturn(Future.successful(
+        notificationRecord.copy(reference = "ref2", status = RecordStatusEnum.NOT_PROCESSED_PENDING_RETRY, updatedAt = LocalDateTime.now())
+      ))
+      val result = await(service.invoke)
+      result.isRight shouldBe true
+      result.getOrElse("fail") shouldBe "Processed all notifications"
+      verify(mockFileNotificationRepository, times(1)).updateFileNotification(Matchers.eq("ref2"), Matchers.eq(RecordStatusEnum.NOT_PROCESSED_PENDING_RETRY))
+    }
+
+    "process the notifications and return Left if some fail" in new Setup {
+      val exception = new Exception("woopsy")
+      when(mockFileNotificationRepository.getPendingNotifications()).thenReturn(Future.successful(pendingNotificationTwo))
+      when(mockFileNotificationRepository.getFilesReceivedBySDES()).thenReturn(Future.successful(pendingNotificationTwo))
+      when(mockFileNotificationRepository.updateFileNotification(Matchers.any(), Matchers.any()))
+        .thenReturn(Future.successful(notificationRecord.copy(reference = "ref2", status = RecordStatusEnum.NOT_PROCESSED_PENDING_RETRY, updatedAt = LocalDateTime.now())))
+        .thenReturn(Future.failed(exception))
+      val result = await(service.invoke)
+      result.isLeft shouldBe true
+      result.left.getOrElse("fail") shouldBe FailedToProcessNotifications
+      withCaptureOfLoggingFrom(logger) {
+        logs => {
+          eventually {
+            logs.exists(_.getMessage.contains(PagerDutyKeys.UNKNOWN_EXCEPTION_FROM_SDES))
+            logs.exists(_.getMessage.contains(PagerDutyKeys.FAILED_TO_PROCESS_FILE_NOTIFICATION))
+          }
+        }
+      }
+    }
+
+    "process the notifications and return Left if all fail" in new Setup {
+      val exception = new Exception("woopsy")
+      when(mockFileNotificationRepository.getPendingNotifications()).thenReturn(Future.successful(pendingNotificationTwo))
+      when(mockFileNotificationRepository.getFilesReceivedBySDES()).thenReturn(Future.successful(pendingNotificationTwo))
+      when(mockFileNotificationRepository.updateFileNotification(Matchers.any(), Matchers.any()))
+        .thenReturn(Future.failed(exception))
+        .thenReturn(Future.failed(exception))
```

Comment:
.thenReturn will always return the same value no matter how many times its called, IIRC. You can remove the duplicate safely